### PR TITLE
6 add messaging

### LINKED ISSUE #6: Connect video calling and messaging together

### PR DESCRIPTION
**Pull request by LemonFace0309**
used webRTC through simple-peer for video/messaging. Not sure if handling client side is the most secure tbh. We'll go with it for now.

closes #6 

image

### CODE REVIEW

File: components/Spaces/Chat/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  const keyPressHandler = (e) => {
+    if (e.keyCode == 13) {
+      submitHandler(e)
+    }
+  }
+
+  return (
+    <>
+      <Paper className="flex flex-col h-96 min-h-full w-9/12 max-w-lg" elevation={3}>
```
Comment (mabelene) [inline components/Spaces/Chat/index.js:33]: waitt I didn't even know you could do that, that's pretty cool

**Review by mabelene — APPROVED** [review]

File: components/Spaces/Chat/index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    const messageDidSend = sendMessage(text)
+    setError(!messageDidSend)
+    setText('')
+    setTimeout(() => {
+      setError(false)
+    }, 2000)
+  }
+
+  const keyPressHandler = (e) => {
+    if (e.keyCode == 13) {
```
Comment (jimmyygit) [inline components/Spaces/Chat/index.js:26]: can we put a constant on this. idk what 13 is

File: context/ConversationProvider.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+// const arrayEquality = (a, b,) => {
+//   if (a.length !== b.length) return false
+
+//   a.sort()
+//   b.sort()
+
+//   return a.every((element, index) => {
+//     return element === b[index]
+//   })
+// }
```
Comment (jimmyygit) [inline context/ConversationProvider.js:69]: do we need this? if maybe, would probably be good to put it in a `utils` module

**Review by jimmyygit — APPROVED** [review]
noice

File: components/Spaces/Chat/index.js (hunk 1 above)
Comment (LemonFace0309) [reply components/Spaces/Chat/index.js:33]: aha thanks

File: context/ConversationProvider.js (hunk 3 above)
Comment (LemonFace0309) [reply context/ConversationProvider.js:69]: naw. I commented out for now because I might need to use it in the future. Will abstract later.

**Pull request merged**